### PR TITLE
Fix version in package.json, support yarn 3

### PR DESCRIPTION
**Pull request by IanVS**
## Change Type

Indicate the type of change your pull request is:

- [X] `maintenance`
- [ ] `documentation`
- [ ] `patch`
- [ ] `minor`
- [ ] `major`

This does a few main things:

- Fixes the version number in package.json (is it not updated as part of the release process?)
- Adds support for yarn 3, otherwise a fresh clone and `yarn install` sets up pnp mode and a lot of extra files.
- Updates the release CI job to:
	- Use updated actions to avoid deprecation warnings
	- Use yarn 3 + cache
	- Only run once a PR is opened to ensure the release bot can update the PR body with the canary version

<details>
  <summary>📦 Published PR as canary version: <code>0.2.2--canary.41.6009636227.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/jest@0.2.2--canary.41.6009636227.0
  # or 
  yarn add @storybook/jest@0.2.2--canary.41.6009636227.0
  ```
</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@storybook/jest",
-  "version": "0.1.1-next.0",
+  "version": "0.2.1",
```
Comment (kasperpeulen) [inline package.json:3]: was this intentional?

**Review by kasperpeulen — COMMENTED** [review]
Nice!

File: package.json (hunk 1 above)
Comment (IanVS) [reply package.json:3]: Yes, isn't that the latest version of this package?  It seems like it isn't being updated by the release script, perhaps.

Comment (kasperpeulen) [reply package.json:3]: I think this branch is supposed to release next versions, and the main branch is releasing stable releases. cc @yannbf

Comment (IanVS) [reply package.json:3]: oh interesting.  And I guess there's nothing that rebases `next` onto `main`?  It feels weird to have `next` with a version lower than `main`, and also keep it as the default branch.  I'm not really familiar with what the correct approach is here, so happy to revert this or whatever @yannbf says is needed.

Comment (yannbf) [reply package.json:3]: Whenever I merge next onto main, it triggers a release script and that bumps the package version. I then merge main onto next, to get changelog and proper version in package.json. I guess with recent fixes from Kasper we forgot to do that.

**Pull request merged**